### PR TITLE
fix(curriculum) - clearer RecordCollection instructions

### LINKED ISSUE #49732: Record Collection
### Describe the Issue

There was a discussion on the forum for an update to the wording here, but we forgot to turn it into an issue.

https://forum.freecodecamp.org/t/give-record-collection-challenge-more-context/587126
https://forum.freecodecamp.org/t/suggestion-for-possible-change-to-record-collection-instructions/598781

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/record-collection

### Your code

N/A

### Expected behavior

Proposed wording update:

---

You are creating a function that aids in the maintenance of a musical album collection. The collection is organized in an object that contains multiple albums which are also objects. Each album is represented in the collection with a unique `id` as the property name. Within each album object, there are various properties describing information about the album. Not all albums have complete information.

The `updateRecords` function takes 4 arguments represented by the following function parameters:

* `records` - an object containing several individual albums
* `id` - a number representing a specific album in the `records` object
* `prop` - a string representing the name of the album’s property to update
* `value` - a string containing the information used to update the album’s property

Complete the function using the rules below to modify the object passed to the function.

* Your function must always return the entire `records` object.
* If `value` is an empty string, delete the given `prop` property from the album.
* If `prop` isn’t `"tracks"` and `value` isn't an empty string, assign the `value` to that album’s `prop`.
* If `prop` is `"tracks"` and `value` isn’t an empty string, add the `value` to the end of the album’s existing `"tracks"` array.
* If album doesn’t have a `"tracks"` property, create a new array for the album's `"tracks"` property before adding the `value` to it.

### Screenshots

_No response_

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeremylt**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #49732

### CODE REVIEW

**Comment by jeremylt** [comment]
Ideally a user should be able to understand or figure out what each function input means, but as we see on the forum, they just aren't quite equipped to do so by the time they reach this problem. These changes encapsulate the most common advice given to questions on the forum about this Challenge.

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/record-collection.md [hunk 1]
```diff
@@ -8,17 +8,24 @@ dashedName: record-collection
 
 # --description--
 
-You are given an object literal representing a part of your musical album collection. Each album has a unique id number as its key and several other properties. Not all albums have complete information.
+You are creating a function that aids in the maintenance of a musical album collection. The collection is organized as an object that contains multiple albums which are also objects. Each album is represented in the collection with a unique `id` as the property name. Within each album object, there are various properties describing information about the album. Not all albums have complete information.
 
-You start with an `updateRecords` function that takes an object literal, `records`, containing the musical album collection, an `id`, a `prop` (like `artist` or `tracks`), and a `value`. Complete the function using the rules below to modify the object passed to the function.
+The updateRecords function takes 4 arguments represented by the following function parameters:
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/record-collection.md:13]:
```suggestion
The `updateRecords` function takes 4 arguments represented by the following function parameters:
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my suggestions and reasoning.

**Comment by jeremylt** [comment]
I replied why I think the two remaining review comments are not the way to go in my opinion, so I think we are back to waiting on review.

**Comment by RandellDawson** [comment]
@jeremylt I understand your intent.  I just wish there was a way to not make it so "wordy".  This is the first challenge where logic order matters but we are still really trying to get the users understanding how to use objects.  Is there a way to present the challenge instructions to guide the user to a specific order of logic?  That should make it easier for the user to just focus on the syntax needed.  Maybe the instructions can be written to explain the logic should be written in the same order as the rules?

**Comment by jeremylt** [comment]
Do we want to that strongly push this challenge into 'exactly replicate our pseudo code'? We can always make more adjustments in the future. I think we can make the changes in this PR and then add even more hand holding if the questions on the forum show that this is still too much of a leap.

**Comment by RandellDawson** [comment]
> Do we want to that strongly push this challenge into 'exactly replicate our pseudo code'?

I think so.  The new project-based curriculum is going to do just that. Plus, this challenge is not part of the Basic Algorithm section, so users at this point should not be required to flesh out the algorithm.  We do it for them with the pseudo code instructions.  In fact, I would even change the instructions to reflect that this part is pseudo code (we mention it a lot but never show an example).  None of this stops users from creating different solutions, but should definitely reduce the huge leap this particular challenge has historically created and also bridge the gap between seeing a full algorithm and having the users implement it.

Personally, I wish the project-based curriculum would do more of developing the algorithms for project use cases in the beginning instead of "do this" and "do that" to create a project.

**Comment by jeremylt** [comment]
Personally I think that's the wrong approach.

Since we are directly of the opposite opinion here, I'm going to wait and only add text to the effect of "you should execute these conditions in this order" along with simplification of the bullet points using the assumption that the bullet points are coded exactly in that order if there is a broader consensus to do so. 

Students often and easily miss parts of the instructions, so I think making the instructions more robust to students missing sentences or putting things in different order is the right decision here.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
> Personally I think that's the wrong approach.

I am not sure what part of what I said is the wrong approach, but I will just approve the PR since I whatever I do here anymore is rejected.  I am done contributing here.

**Pull request merged**